lib/licensed/sources/npm.rb

```diff
@@ -135,6 +137,18 @@ def yarn_lock_present
       def include_non_production?
         config.dig("npm", "production_only") == false
       end
+
+      def missing_peer?(parent, dependency, name)
+        dependency["peerMissing"] || (dependency["missing"] && peer_dependency(parent, name))
+      end
+
+      def peer_dependency(parent, name)
+        parent.dig("peerDependencies", name)
+      end
+
+      def extract_version(parent, name)
+        parent.dig("_dependencies", name) || peer_dependency(parent, name)
+      end
```

Comment:
can you add comments about the expected return values from each of these functions?  It was a little confusing that `peer_dependency` returned a usable version value, maybe that function could also be renamed?